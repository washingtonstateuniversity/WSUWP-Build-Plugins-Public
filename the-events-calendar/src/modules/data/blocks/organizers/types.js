/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_CLASSIC_ORGANIZERS = `${ PREFIX_EVENTS_STORE }/SET_CLASSIC_ORGANIZERS`;
export const REMOVE_CLASSIC_ORGANIZERS = `${ PREFIX_EVENTS_STORE }/REMOVE_CLASSIC_ORGANIZERS`;
export const ADD_CLASSIC_ORGANIZERS = `${ PREFIX_EVENTS_STORE }/ADD_CLASSIC_ORGANIZERS`;

export const ADD_ORGANIZER_BLOCK = `${ PREFIX_EVENTS_STORE }/ADD_ORGANIZER_BLOCK`;
export const REMOVE_ORGANIZER_BLOCK = `${ PREFIX_EVENTS_STORE }/REMOVE_ORGANIZER_BLOCK`;
