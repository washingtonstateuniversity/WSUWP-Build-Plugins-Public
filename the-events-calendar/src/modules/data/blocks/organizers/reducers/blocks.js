/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';
import block from './block';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
		case types.REMOVE_ORGANIZER_BLOCK:
			return {
				...state,
				[ action.payload.id ]: block( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_ORGANIZER_BLOCK:
			return [ ...state ].filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
