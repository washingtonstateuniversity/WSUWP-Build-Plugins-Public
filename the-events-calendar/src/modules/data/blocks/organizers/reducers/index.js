export { default as classic } from './classic';
export { default as blocks } from './blocks';
