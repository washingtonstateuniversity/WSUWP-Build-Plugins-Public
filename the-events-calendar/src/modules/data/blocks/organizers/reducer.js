/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import { classic, blocks } from './reducers';

export default combineReducers( {
	blocks,
	classic,
} );
