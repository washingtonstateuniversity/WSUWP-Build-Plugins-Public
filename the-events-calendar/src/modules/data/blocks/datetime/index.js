/**
 * Internal dependencies
 */
import reducer from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as thunks from './thunks';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, thunks, selectors, sagas };
