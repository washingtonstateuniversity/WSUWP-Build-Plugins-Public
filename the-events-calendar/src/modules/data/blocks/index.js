/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
