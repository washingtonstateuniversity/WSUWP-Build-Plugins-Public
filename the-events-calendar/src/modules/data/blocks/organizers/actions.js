/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events/data/blocks/organizers';

export const setOrganizersInClassic = ( organizers ) => ( {
	type: types.SET_CLASSIC_ORGANIZERS,
	payload: {
		organizers,
	},
} );

export const removeOrganizerInClassic = ( id ) => ( {
	type: types.REMOVE_CLASSIC_ORGANIZERS,
	payload: {
		id,
	},
} );

export const addOrganizerInClassic = ( organizer ) => ( {
	type: types.ADD_CLASSIC_ORGANIZERS,
	payload: {
		organizer,
	},
} );

export const addOrganizerInBlock = ( id, organizer ) => ( {
	type: types.ADD_ORGANIZER_BLOCK,
	payload: {
		id,
		organizer,
	},
} );

export const removeOrganizerInBlock = ( id, organizer ) => ( {
	type: types.REMOVE_ORGANIZER_BLOCK,
	payload: {
		id,
		organizer,
	},
} );
