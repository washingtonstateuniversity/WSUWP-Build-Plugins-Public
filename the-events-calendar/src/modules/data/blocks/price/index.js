/**
 * Internal dependencies
 */
import reducer from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import sagas from './sagas';

export default reducer;
export { selectors, actions, types, sagas };
