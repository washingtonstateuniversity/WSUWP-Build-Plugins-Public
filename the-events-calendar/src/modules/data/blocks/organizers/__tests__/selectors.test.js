/**
 * Internal dependencies
 */
import { selectors } from '@moderntribe/events/data/blocks/organizers';

const state = {
	events: {
		blocks: {
			organizers: {
				blocks: {
					byId: {
						firstBlock: {
							organizer: 100,
						},
						secondBlock: {
							organizer: 101,
						},
					},
					allIds: [ 100, 101 ],
				},
				classic: [ 98, 99, 100 ],
			},
		},
	}
};

describe( '[STORE] - Organizers selectors', () => {
	it( 'Should return the classic organizers', () => {
		expect( selectors.getOrganizersInClassic( state ) ).toEqual( [ 98, 99, 100 ] );
	} );

	it( 'Should return the organizer block', () => {
		expect( selectors.getOrganizerBlock( state, { clientId: 'firstBlock' } ) )
			.toEqual( state.events.blocks.organizers.blocks.byId.firstBlock );
		expect( selectors.getOrganizerBlock( state, { clientId: 'secondBlock' } ) )
			.toEqual( state.events.blocks.organizers.blocks.byId.secondBlock );
		expect( selectors.getOrganizerBlock( state, { clientId: 'thirdBlock' } ) )
			.toEqual( undefined );
	} );

	it( 'Should return the organizer in a block', () => {
		expect( selectors.getOrganizerInBlock( state, { clientId: 'firstBlock' } ) ).toEqual( 100 );
		expect( selectors.getOrganizerInBlock( state, { clientId: 'secondBlock' } ) ).toEqual( 101 );
		expect( selectors.getOrganizerInBlock( state, { clientId: 'thirdBlock' } ) ).toEqual( undefined );
	} );

	it( 'Should return the organizers in a block', () => {
		expect( selectors.getOrganizersInBlock( state ) ).toEqual( [ 100, 101 ] );
	} );

	it( 'Should return the mapped organizers', () => {
		expect( selectors.getMappedOrganizers( state ) ).toMatchSnapshot();
	} );
} );
