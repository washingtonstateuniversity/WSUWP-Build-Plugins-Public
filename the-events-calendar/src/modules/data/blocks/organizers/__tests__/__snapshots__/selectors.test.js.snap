// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Organizers selectors Should return the mapped organizers 1`] = `
Array [
  Object {
    "block": false,
    "id": 98,
  },
  Object {
    "block": false,
    "id": 99,
  },
  Object {
    "block": true,
    "id": 100,
  },
]
`;
